src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,947 @@
+            diagnoser_output.append("An exception has occured while performing the outbound connectivity check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+        logger.warning("An exception has occured while performing the outbound connectivity check on the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Outbound_Connectivity_Check_Fault_Type, summary="Error occured while performing outbound connectivity check in the cluster")
+        diagnoser_output.append("An exception has occured while performing the outbound connectivity check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return consts.Diagnostic_Check_Incomplete, storage_space_available
+
+
+def check_msi_certificate_presence(corev1_api_instance):
+
+    global diagnoser_output
+    try:
+
+        # Initializing msi certificate as not present
+        msi_cert_present = False
+
+        # Going thorugh all the secrets in azure-arc
+        all_secrets_azurearc = corev1_api_instance.list_namespaced_secret(namespace="azure-arc")
+        for secrets in all_secrets_azurearc.items:
+            # If name of secret is azure-identity-certificate then we stop there
+            if(secrets.metadata.name == consts.MSI_Certificate_Secret_Name):
+                msi_cert_present = True
+
+        # Checking if msi cerificate is present or not
+        if not msi_cert_present:
+            print("Error: Unable to pull MSI certificate. Please ensure to meet the following network requirements 'https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/quickstart-connect-cluster?tabs=azure-cli#meet-network-requirements'. \n")
+            diagnoser_output.append("Error: Unable to pull MSI certificate. Please ensure to meet the following network requirements 'https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/quickstart-connect-cluster?tabs=azure-cli#meet-network-requirements'. \n")
+            return consts.Diagnostic_Check_Failed
+
+        return consts.Diagnostic_Check_Passed
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+        logger.warning("An exception has occured while performing the msi certificate check on the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.MSI_Cert_Check_Fault_Type, summary="Error occurred while trying to perform MSI ceritificate presence check")
+        diagnoser_output.append("An exception has occured while performing the msi certificate check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return consts.Diagnostic_Check_Incomplete
+
+
+def check_cluster_security_policy(corev1_api_instance, helm_client_location, release_namespace):
```

Comment:
we should add "probable" here as well since this is also an assumption based on the fact that KAP is absent and cluster connect is enabled

Reply:
check_probable_presence_of_pod_security_policies